File: backend/app/api/v1/schemas/document.py
from datetime import datetime
from typing import Any, Dict, List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class DocumentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    title: str
    filename: str
    content_type: str
    storage_path: str
    metadata_info: Optional[Dict[str, Any]] = None
    created_at: datetime


class DocumentDetail(DocumentRead):
    model_config = ConfigDict(from_attributes=True)
    text: str


class DocumentSearchRequest(BaseModel):
    query: str
    limit: int = 5
    document_id: Optional[UUID] = None


class DocumentChunkResult(BaseModel):
    chunk_id: UUID
    document_id: UUID
    document_title: str
    document_filename: str
    chunk_index: int
    content: str


class DocumentSearchResponse(BaseModel):
    results: List[DocumentChunkResult]


class DocumentChatRequest(BaseModel):
    query: str
    limit: int = 5
    document_id: Optional[UUID] = None


class DocumentChatResponse(BaseModel):
    answer: str
    sources: List[DocumentChunkResult]

File: frontend/src/types/document.ts
type Document = {
    id: string;
    user_id: string;
    title: string;
    filename: string;
    content_type: string;
    storage_path: string;
    text: string;
    metadata_info?: Record<string, any>;
    created_at: string;
}

type CreateDocument = {
    title: string,
    file: File
}

export type { Document, CreateDocument }